pkg/csv2lp/lp_reader.go

```diff
@@ -21,24 +25,35 @@ func LineProtocolFilter(reader io.Reader) *LineProtocolFilterReader {
 	lineReader.LineNumber = 1 // start counting from 1
 	return &LineProtocolFilterReader{
 		lineReader: lineReader,
+		line:       make([]byte, 4096),
+		wrapBuf:    make([]byte, 4096),
 	}
 }
 
 func (state *LineProtocolFilterReader) Read(b []byte) (int, error) {
-	for {
-		bytesRead, err := state.lineReader.Read(b)
-		if err != nil {
-			return bytesRead, err
-		}
-		state.LineNumber = state.lineReader.LastLineNumber
-		buf := b[0:bytesRead]
-		pts, err := models.ParsePoints(buf) // any time precision because we won't actually use this point
-		if err != nil {
-			log.Printf("invalid point on line %d: %v\n", state.LineNumber, err)
-			continue
-		} else if len(pts) == 0 { // no points on this line
-			continue
+	bytesRead, err := state.lineReader.Read(state.line)
+	if err != nil && bytesRead == 0 {
+		return 0, err
```

Comment:
This conditional changes the behavior; in the case where bytes are read, but an error is returned (perhaps EOF from a final line without a newline), this will continue, whereas the existing code will return the bytesRead and the error.

Reply:
Is this a fix for a problem you found, or an inadvertent change in behavior?